### ISSUE TITLE
win_chocolatey: installing kodi breaks playbook run the first time

### ISSUE THREAD

**Issue by dreamcat4** [ansible/ansible-modules-extras#2813]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

win_chocolatey
##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.2.0 (devel 58b9f637a1) last updated 2016/08/24 09:07:32 (GMT +100)
  lib/ansible/modules/core: (detached HEAD 368ca738fa) last updated 2016/08/23 20:15:23 (GMT +100)
  lib/ansible/modules/extras: (detached HEAD 0749ce6faa) last updated 2016/08/23 20:15:24 (GMT +100)
  config file = /home/id/ansible/ansible.cfg
  configured module search path = ['./library']
```
##### OS / ENVIRONMENT

Windows 10 Pro anniversary update edition Build 14393.
##### SUMMARY

On a clean system, the first time kodi 16.1 is installed via win_chocolatey, throws some exception. And the playbook run cannot recover, even with ignore_errors.

Subsequent invocations show the package is already installed. (no error).
##### STEPS TO REPRODUCE

Clean system, windows 10 pro.

```
- name: Install Kodi
  win_chocolatey: state=present name=kodi version=16.1
  ignore_errors: yes
```
##### EXPECTED RESULTS

If it fails, then the playbook run should continue.
##### ACTUAL RESULTS

```
 ______________________________
< TASK [4_pkgs : Install Kodi] >
 ------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

task path: /home/id/ansible/windows/roles/4_pkgs/tasks/kodi.yml:14
Traceback (most recent call last):
  File "/home/id/dev/ansible/lib/ansible/plugins/connection/winrm.py", line 274, in exec_command
    result = self._winrm_exec(cmd_parts[0], cmd_parts[1:], from_exec=True)
  File "/home/id/dev/ansible/lib/ansible/plugins/connection/winrm.py", line 236, in _winrm_exec
    self.protocol.cleanup_command(self.shell_id, command_id)
  File "/usr/local/lib/python2.7/dist-packages/winrm/protocol.py", line 307, in cleanup_command
    res = self.send_message(xmltodict.unparse(req))
  File "/usr/local/lib/python2.7/dist-packages/winrm/protocol.py", line 207, in send_message
    return self.transport.send_message(message)
  File "/usr/local/lib/python2.7/dist-packages/winrm/transport.py", line 173, in send_message
    response = self.session.send(prepared_request, timeout=self.read_timeout_sec)
  File "/usr/local/lib/python2.7/dist-packages/requests/sessions.py", line 585, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/requests/adapters.py", line 459, in send
    raise ConnectTimeout(e, request=request)
ConnectTimeout: HTTPSConnectionPool(host='brix-3205', port=5986): Max retries exceeded with url: /wsman (Caused by ConnectTimeoutError(<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f33163c8c90>, 'Connection to brix-3205 timed out. (connect timeout=30)'))
fatal: [brix-3205]: UNREACHABLE! => {"changed": false, "msg": "failed to exec cmd PowerShell -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -EncodedCommand UwBlAHQALQBTAHQAcgBpAGMAdABNAG8AZABlACAALQBWAGUAcgBzAGkAbwBuACAATABhAHQAZQBzAHQACgBUAHIAeQAKAHsACgAmACAAJwBDADoAXABVAHMAZQByAHMAXABhAG4AcwBpAGIAbABlAFwAQQBwAHAARABhAHQAYQBcAEwAbwBjAGEAbABcAFQAZQBtAHAAXABhAG4AcwBpAGIAbABlAC0AdABtAHAALQAxADQANwAyADUAMAA3ADQANAA2AC4ANgA2AC0AMQA1ADYAOQAwADMANAAyADQAOQAzADgANwAzADcAXAB3AGkAbgBfAGMAaABvAGMAbwBsAGEAdABlAHkALgBwAHMAMQAnAAoAfQAKAEMAYQB0AGMAaAAKAHsACgAkAF8AbwBiAGoAIAA9ACAAQAB7ACAAZgBhAGkAbABlAGQAIAA9ACAAJAB0AHIAdQBlACAAfQAKAEkAZgAgACgAJABfAC4ARQB4AGMAZQBwAHQAaQBvAG4ALgBHAGUAdABUAHkAcABlACkACgB7AAoAJABfAG8AYgBqAC4AQQBkAGQAKAAnAG0AcwBnACcALAAgACQAXwAuAEUAeABjAGUAcAB0AGkAbwBuAC4ATQBlAHMAcwBhAGcAZQApAAoAfQAKAEUAbABzAGUACgB7AAoAJABfAG8AYgBqAC4AQQBkAGQAKAAnAG0AcwBnACcALAAgACQAXwAuAFQAbwBTAHQAcgBpAG4AZwAoACkAKQAKAH0ACgBJAGYAIAAoACQAXwAuAEkAbgB2AG8AYwBhAHQAaQBvAG4ASQBuAGYAbwAuAFAAbwBzAGkAdABpAG8AbgBNAGUAcwBzAGEAZwBlACkACgB7AAoAJABfAG8AYgBqAC4AQQBkAGQAKAAnAGUAeABjAGUAcAB0AGkAbwBuACcALAAgACQAXwAuAEkAbgB2AG8AYwBhAHQAaQBvAG4ASQBuAGYAbwAuAFAAbwBzAGkAdABpAG8AbgBNAGUAcwBzAGEAZwBlACkACgB9AAoARQBsAHMAZQBJAGYAIAAoACQAXwAuAFMAYwByAGkAcAB0AFMAdABhAGMAawBUAHIAYQBjAGUAKQAKAHsACgAkAF8AbwBiAGoALgBBAGQAZAAoACcAZQB4AGMAZQBwAHQAaQBvAG4AJwAsACAAJABfAC4AUwBjAHIAaQBwAHQAUwB0AGEAYwBrAFQAcgBhAGMAZQApAAoAfQAKAFQAcgB5AAoAewAKACQAXwBvAGIAagAuAEEAZABkACgAJwBlAHIAcgBvAHIAXwByAGUAYwBvAHIAZAAnACwAIAAoACQAXwAgAHwAIABDAG8AbgB2AGUAcgB0AFQAbwAtAEoAcwBvAG4AIAB8ACAAQwBvAG4AdgBlAHIAdABGAHIAbwBtAC0ASgBzAG8AbgApACkACgB9AAoAQwBhAHQAYwBoAAoAewAKAH0ACgBFAGMAaABvACAAJABfAG8AYgBqACAAfAAgAEMAbwBuAHYAZQByAHQAVABvAC0ASgBzAG8AbgAgAC0AQwBvAG0AcAByAGUAcwBzACAALQBEAGUAcAB0AGgAIAA5ADkACgBFAHgAaQB0ACAAMQAKAH0ACgBGAGkAbgBhAGwAbAB5ACAAewAgAFIAZQBtAG8AdgBlAC0ASQB0AGUAbQAgACIAQwA6AFwAVQBzAGUAcgBzAFwAYQBuAHMAaQBiAGwAZQBcAEEAcABwAEQAYQB0AGEAXABMAG8AYwBhAGwAXABUAGUAbQBwAFwAYQBuAHMAaQBiAGwAZQAtAHQAbQBwAC0AMQA0ADcAMgA1ADAANwA0ADQANgAuADYANgAtADEANQA2ADkAMAAzADQAMgA0ADkAMwA4ADcAMwA3ACIAIAAtAEYAbwByAGMAZQAgAC0AUgBlAGMAdQByAHMAZQAgAC0ARQByAHIAbwByAEEAYwB0AGkAbwBuACAAUwBpAGwAZQBuAHQAbAB5AEMAbwBuAHQAaQBuAHUAZQAgAH0A", "unreachable": true}
    to retry, use: --limit @windows/gibbos-brix.retry
 ____________
< PLAY RECAP >
 ------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

brix-3205                  : ok=326  changed=182  unreachable=1    failed=0   
```

**Comment by ansibot** [comment]
@smadam813 ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dreamcat4** [comment]
So would wrapping inside of a `block` be able to rescue it from this type of exception? I don't mind if there is an error, just would like to recover from it.

**Comment by ansibot** [comment]
@smadam813, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@smadam813, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@smadam813, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@smadam813, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by dagwieers** [comment]
@dreamcat4 We have rewritten the win_chocolatey partly. So please test your issue with the v2.3 release, or the devel branch. And report back :-) If this issue is still present, please move it to https://github.com/ansible/ansible

Thanks in advance !

**Comment by dagwieers** [comment]
A lot of the supporting code for Windows/WinRM support has been rewritten. And we believe this issue might be resolved. If you still experience issues using v2.3 or the devel branch, we would like to hear from you. Please check ansible/ansible#23320 or open a new issue at: https://github.com/ansible/ansible/issues

Thanks !
